### ISSUE TITLE
RFE: defaults: higher max mem/vcpus so live increase works out of the box

### ISSUE THREAD

**Issue by Lennie** [virt-manager/virt-manager#141]
For KVM/Qemu it probably makes sense for new VMs to just have the physical limits as the maximum not the current allocation.

If you create a new VM and want to hot-add CPU or memory you'll need to stop/start the VM. Even though hot-plug would have just worked with virt-manager, KVM/Qemu and the VM (tested with some what recent Linux host and VM).

Even when the maximum is set higher and current allocation is smaller than the physical host than the VM still boots. So it might even be possible to still do live migration, but I didn't test it.

**Comment by crobinso** [comment]
Thanks for the suggestion. It's not that simple though. If you have memory=16g and currentMemory=4g, this only does the correct thing if the guest has a working memballoon driver. So out of the box that excludes windows. For Linux it may work.

I don't know if there's implications for vcpu defaults.

I'm planning to kick off a round of discussion with virt developers about improving virt-manager/virt-install defaults in a few areas, I will ask about this suggestion, thanks.

**Comment by crobinso** [comment]
I looked into the vcpu maxvcpus case and I'm not confident it's something we can unconditionally set without possible problems. The maxvcpus value is communicated to the guest via ACPI and if someone uses numa it requires describing a numa topology for the full maxvcpus value, vs just what vcpus are actually attached. Memory has the windows vs linux problem I described and requires guest communication to not oversubscribe memory.

Basically I think for virt-manager/virt-install's needs this is too risky to try to set up unconditionally, so we won't be doing it.
